Automattic/newspack-newsletters
# Add/active campaign tags

### All Submissions:


### Changes proposed in this Pull Request:



Adds support to edit contacts tags in Active Campaign provider

### How to test the changes in this Pull Request:

1. Setup Active Campaign as your provider
2. Open the CLI terminal `wp shell`
3. Run the following commands
```
wp> $cc = [ 'name' => 'Test contact', 'email' => 'test@test.com', 'metadata' => [ 'current_page_url' => home_url() ], 'tags' => [ 'testing']  ];
wp> $list_id = 'xxxx'; // replace with your List ID
wp> Newspack_Newsletters_Subscription::add_contact( $cc, $list_id);
```
4. Visit your AC dashboard and confirm the contact was created and the tag was added
5. Now test editing existing contacts:

```
wp> $ac = Newspack_Newsletters_Active_Campaign::instance();
wp> $ac->update_contact_tags('test@test.com', ['test2'] );
wp> $ac->update_contact_tags('test@test.com', ['add2'], ['test2'] ); // third param remove tags
```

6. Visit your AC dashboard and confirm each action was successfully performed.

### Other information:
